mono/metadata/mempool.c

```diff
@@ -254,6 +274,102 @@ get_next_size (MonoMemPool *pool, int size)
 #endif
 
 /**
+ * mono_mempool_enable_canaries:
+ *
+ * Enable placing canaries between allocations
+ */
+
+void
+mono_mempool_enable_canaries (void)
+{
+	canary_count = DEFAULT_ENABLED_CANARY_COUNT;
+	return;
+}
+
+/**
+ * mono_mempool_check_canaries:
+ *
+ * Check that our canaries are still alive.
+ */
+
+void
+mono_mempool_check_canaries (MonoMemPool *pool)
+{
+	if (canary_count == 0)
+		return;
+
+#ifdef MALLOC_ALLOCATION
+	for (Chunk* c = pool->chunks; c != NULL; c++) {
+		gpointer start = c + sizeof(Chunk) + c->size;
+		gpointer end = start + PADDING_COUNT;
+
+		// The end should be the value stored at the start of the canary
+		assert(end == start + PADDING_COUNT + 1);
```

Comment:
I can't see how this would ever be true given the above definition of end.

Please use g_assert instead of assert. And add a space before the open parens.